### PR TITLE
Modify rule S1206: LaYC format

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**

### CODE REVIEW

File: rules/S1206/java/rule.adoc [hunk 1]
```diff
@@ -1,10 +1,56 @@
 == Why is this an issue?
 
-include::../description.adoc[]
```
Comment (gregory-paidis-sonarsource) [inline rules/S1206/java/rule.adoc:3]: Description, noncompliant and compliant were used only here, so I inlined them.

File: rules/S1206/java/rule.adoc [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+According to the Java Language Specification, there is a contract between ``++equals(Object)++`` and ``++hashCode()++``:
 
-include::../noncompliant.adoc[]
 
-include::../compliant.adoc[]
+____
+If two objects are equal according to the ``++equals(Object)++`` method, then calling the ``++hashCode++`` method on each of the two objects must produce the same integer result.
```
Comment (mary-georgiou-sonarsource) [inline rules/S1206/java/rule.adoc:7]: I guess java part is out of scope right? I'm not reviewing this?

File: rules/S1206/csharp/rule.adoc [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+There is a contract between https://learn.microsoft.com/en-us/dotnet/api/system.object.equals[Object.Equals] and https://learn.microsoft.com/en-us/dotnet/api/system.object.gethashcode[Object.GetHashCode]: If two objects are equal according to the `Equals` method, then calling `GetHashCode` on each of them must yield the same integer. If this is not the case, many collections, such as a https://learn.microsoft.com/en-us/dotnet/api/system.collections.hashtable[Hashtable] or a https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2[Dictionary] won't handle class instances correctly.
 
+In order to comply with the contract, `Equals` and `GetHashCode` should be either both inherited, or both overridden.
 
-In order to comply with the contract, ``++Equals(object)++`` and ``++GetHashCode()++`` should be either both inherited, or both overridden.
+== How to fix it
 
-=== Noncompliant code example
+When you want to use `Equals` and `GetHashCode`, either override both of them, or simply inherit them.
```
Comment (mary-georgiou-sonarsource) [inline rules/S1206/csharp/rule.adoc:9]:
```suggestion
When you override `Equals` then you have to override also `GetHashCode`. You have to override both of them, or simply inherit them.
```

**Review by mary-georgiou-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S1206/java/rule.adoc (hunk 2 above)
Comment (gregory-paidis-sonarsource) [reply rules/S1206/java/rule.adoc:7]: you could look that everything looks correct, I just moved stuff around.

File: rules/S1206/csharp/rule.adoc [hunk 4]
```diff
@@ -36,9 +41,18 @@ class MyClass
...
+
+* https://cwe.mitre.org/data/definitions/581[MITRE, CWE-581] - Object Model Violation: Just One of Equals and Hashcode Defined
+* https://learn.microsoft.com/en-us/dotnet/api/system.object.equals[Object.Equals Method] 
+* https://learn.microsoft.com/en-us/dotnet/api/system.object.gethashcode[Object.GetHashCode Method]
+* https://learn.microsoft.com/en-us/dotnet/api/system.collections.hashtable[Hashtable class]  
+* https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2[Dictionary<TKey,TValue> Class] 
+
 
-ifdef::env-github,rspecator-view[]
+ifdef::env-github,rspecator-view,env-vscode[]
```
[the remark this answers is not in the document]
Comment (gregory-paidis-sonarsource) [reply rules/S1206/csharp/rule.adoc:55]: By using this flag, you can render this part locally in vscode. It works like an environment variable, useful for development, no impact in the product.

[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply rules/S1206/csharp/rule.adoc:55]: Cool thanks - I'll use it too.

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**